diff -ur openbsd-nopatch/usr.bin/signify/crypto_api.c b/usr.bin/signify/crypto_api.c
--- a/usr.bin/signify/crypto_api.c	2014-01-08 05:59:46.000000000 +0200
+++ b/usr.bin/signify/crypto_api.c	2020-10-16 14:29:11.994722510 +0300
@@ -3,6 +3,7 @@
  * Public domain. Author: Ted Unangst <tedu@openbsd.org>
  * API compatible reimplementation of functions from nacl
  */
+#include <sys/cdefs.h>
 #include <sys/types.h>
 
 #include <string.h>
diff -ur a/usr.bin/signify/signify.c b/usr.bin/signify/signify.c
--- a/usr.bin/signify/signify.c	2020-01-21 15:13:21.000000000 +0300
+++ b/usr.bin/signify/signify.c	2020-10-16 14:27:02.856481664 +0300
@@ -14,6 +14,7 @@
  * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
+#include <sys/cdefs.h>
 #include <sys/stat.h>
 
 #include <netinet/in.h>
diff -ur a/usr.bin/signify/zsig.c b/usr.bin/signify/zsig.c
--- a/usr.bin/signify/zsig.c	2019-12-22 09:37:25.000000000 +0300
+++ b/usr.bin/signify/zsig.c	2020-10-16 14:31:00.168365855 +0300
@@ -16,6 +16,8 @@
  */
 
 #ifndef VERIFYONLY
+#include <sys/cdefs.h>
+#include <sys/compat.h>
 #include <stdint.h>
 #include <err.h>
 #include <stdio.h>
